File: Java Programs/insertArray.java
/* JAVA PROGRAM TO INSERT AN ELEMENT
 AT SPECIFIC INDEX IN ARRAY */
 import java.util.*;
 public class insertArray
 {
 	public static void main(String[] args) 
 	{
 		Scanner sc=new Scanner(System.in);
 		System.out.println("Enter Total Array Elements=");
 		int size=sc.nextInt();
 		int []arr=new int[size+1];

 		System.out.println("Enter Elements");
 		for(int i=0;i<arr.length-1;i++)
 		{
 			arr[i]=sc.nextInt();
 		}

 		System.out.println("Elements Before Inserting are :-");
 		for(int i=0;i<arr.length;i++)
 		{
 			System.out.println(+arr[i]+" ");
 		}

 		System.out.println("Enter \"location\" at which you want to insert new element=");
 		int loc=sc.nextInt();
 		System.out.println("Enter \"value\" to be inserted=");
 		int value=sc.nextInt();

 		for(int i=arr.length-1;i>loc;i--)
 		{
 			arr[i]=arr[i-1];
 		}
 		arr[loc]=value;

 		System.out.println("Array after insertion of "+value+" at location "+loc+" is=");
 		for(int i=0;i<arr.length;i++)
 		{
 			System.out.println(" "+arr[i]+" ");
 		}
 	}
 }

File: C++ Programs/insertArray.cpp
#include<iostream>
using namespace std;
int main()
{
	int size;
	cout<<"Enter Size Of Array=";
	cin>>size;
	int newsize=size+1;
	int arr[newsize];
	cout<<"Enter Elements of Array :-"<<endl;
	for(int i=0;i<size;i++)
	{
		cout<<"Enter "<<i<<"th element :-";
		cin>>arr[i];
	}
	cout<<"Array Elements Before Insertion are :-"<<endl;
	for(int i=0;i<newsize;i++)
	{
		cout<<i<<"th elements="<<arr[i]<<endl;
	}
	int loc,value;
	cout<<"Enter \"Location\" where you want to insert new element=";
	cin>>loc;
	cout<<"Enter \"Value\" which you want to insert=";
	cin>>value;
	for(int i=newsize;i>loc;i--)
	{
		arr[i]=arr[i-1];
	}
	arr[loc]=value;

	cout<<"Elements After Insertion are :-"<<endl;
	for(int i=0;i<newsize;i++)
	{
		cout<<i<<"th elements="<<arr[i]<<endl;
	}
	return 0;
}